### DISCUSSION TITLE
Looking for a way to return results of individual folds from compare_models

### DISCUSSION THREAD

**Discussion by bittnerma** [pycaret/pycaret#3701] [Q&A] [answered]
Hi all, 

I'm working with pycaret for the first time and I like it very much. 
However, I'm struggling to find out whether there is a way to get the results per fold per model from the `compare_models` function.

To clarify, if I run the `create_model` function I'm getting all metrics per fold including the standard deviation and mean of those results. If I'm running `compare_models` I only get the mean of all cross-validation results per model, however, under the hood the per-fold-results must also be calculated. Is there a way to access these per-fold individual results when running the 'compare_models' function? (Alternatively knowing the standard deviation of them would help as well.)

I've looked through the documentation, FAQs and did a few google searches for this, but can't seem to find a good solution.
I'm currently just running `create_model` for each model in `s.models()` to get the CV results, but that feels like a very roundabout way to get this. 

Any help is appreciated!

Cheers!

**Comment by mateusrn** [comment]
Hi, thank you for posting this question. 

I'm very much looking forward to hearing from contributors. The average of metrics across folds is hardly ever enough to compare models.

Is there any way to access per-fold results?

**Comment by celestinoxp** [comment]
@ngupta23 Do you want to respond?

**Comment by moezali1** [comment, marked as answer]
There is no way to retrieve by fold metrics from `compare_models`. If you need metric by fold `create_model` function can give you that. You can run `create_model` in a loop and retrieve by fold results from a dataframe.
